### DISCUSSION TITLE
ILI9225 connection to LILYGO T22_V1.1

### DISCUSSION THREAD

**Discussion by dickdevries** [dl9rdz/rdz_ttgo_sonde#345] [Q&A]
Hi,

I'm running software devel20230104 with a OLED 0.9. using VCC GND and SCL on 22 and SDA on 21 and running perfect.
Just to test, I like to connect a ILI9225 TFT but am not sure how to connect, because the hardware configuration info is confusing for me:
TFT SDA = 21
TFT CLK = 22
TFT RST = 16
TFT RS = 0
TFT CS = 0
VCC
GND 

VCC and GND are clear for me, but RS and CS not.

Any help is very appreciated.
73 Dick  DL1CLM

**Comment by LY3OO** [comment]
https://lutz-baer.homepage.t-online.de/wordpress/?p=1164

**Reply by dickdevries** [reply]
Thanks for the link.
I looked at a few Youtube's but they all use different connection numbers, that's confusing me.
![image](https://user-images.githubusercontent.com/33785543/212534938-525a8a89-bd09-4a30-b98a-06e027840841.png)

the board I use is different from the one in your link.
the fingerprint of my board is:
![image](https://user-images.githubusercontent.com/33785543/212534997-d069f6d3-c5e9-4367-ba2b-e9c509e1bf27.png)

Do I understand it correctly that I have to change the RS and CS value in de config file?

73,  Dick  DL1CLM

**Reply by dickdevries** [reply]
Are RS and CS value 0 because at this moment the display type 0 is configured and an OLED display is connected?

**Reply by dl9rdz** [reply]
The recommended configuration is the one described here: https://github.com/dl9rdz/rdz_ttgo_sonde/wiki/Supported-displays
i.e. "Preferred connection on T-Beam 1.0 and 1.1 is SDA=4, SCL=13, RST=14, RS=2, CS=0"

You can set these values manually in the config. But also, it should be configured this way automatically on first startup (only on a board that has been flashed freshly. Once you have configured the board manually once (i.e. saved to config file), it will not do autoconfiguration automatically anymore.)

There is a different wiring for the T-Beam 0.7 (SDA=4, CLK=21, RST=22, RS=2, CS=0). This will NOT work on the T-Beam 1.0 because the v1.0 has a power management chip connected to pins 21 and 22. The OLED can be on the same two pins, because both use the I2C protocol, but OLED with I2C and TFT with SPI cannot share the same pins. So don't use pin 21/22.

I recommend that you use the recommended connection above. Some variations might work, but then (a) you have to manually change to config file and (b) you have to make sure on your own that there are no conflicts with other (internal) uses of the pins.

**Reply by dl9rdz** [reply]
(Note that Lutz' web page uses the smaller T3 V1_6, not the T-Beam. The V1_6 does not have a power management chip (or at least not the complex AXP192 that is programmable via iC2), and thus the connection on Pin 21/22 is fine.)

**Reply by dickdevries** [reply]
Many Thanks Hansi, 
I will use the first option you mentioned:  "Preferred connection on T-Beam 1.0 and 1.1 is SDA=4, SCL=13, RST=14, RS=2, CS=0"
after disconnecting the OLED and will change the Display type to "1".
My LILYGO board is new and I did the first flash, the OLED was already mounted, so now I understand why both values RS and CS are 0 in the config.

thanks, Dick  DL1CLM
